libeos-payg/manager.c

```diff
@@ -739,12 +738,8 @@ set_expiry_time (EpgManager *self,
    * adjustments have no effect */
   guint64 old_expiry_time_secs = self->expiry_time_secs;
 
-  /* If the old PAYG code had expired, start from @now_secs; otherwise start
-   * from the current expiry time (in the future). */
-  guint64 base_secs = (now_secs < old_expiry_time_secs) ? old_expiry_time_secs : now_secs;
-
   /* Clamp to the end of time instead of overflowing. */
-  self->expiry_time_secs = (base_secs <= G_MAXUINT64 - span_secs) ? base_secs + span_secs : G_MAXUINT64;
+  self->expiry_time_secs = (now_secs <= G_MAXUINT64 - span_secs) ? now_secs + span_secs : G_MAXUINT64;
```

Comment:
Not that you necessarily want to change the code now, but you can use [`g_uint64_checked_add`](https://docs.gtk.org/glib/func.uint64_checked_add.html) for these types of things:

```
if (!g_uint64_checked_add (&self->expiry_time_secs, now_secs, span_secs))
  self->expiry_time_secs = G_MAXUINT64;
```

Reply:
I think this is worth changing just for readability

Reply:
Personally, I would have to look that function up while reading the code, but I agree that the suggestion is more readable once you know it. I'm adding a fixup commit.